src/Illuminate/Collections/Enumerable.php

```diff
@@ -808,6 +810,30 @@ public function slice($offset, $length = null);
      */
     public function split($numberOfGroups);
 
+    /**
+     * Get the first item in the collection, but only if exactly
+     * item exists. Otherwise, throw an exception.
+     *
+     * @param  callable|null  $callback
+     * @return mixed
+     *
+     * @throws ItemNotFoundException
+     * @throws MultipleItemsFoundException
+     */
+    public function sole(callable $callback = null);
+
+    /**
+     * Get the first item by the given key value pair, but only if
+     * exactly one item matches the criteria. Otherwise, throw
+     * an exception.
+     *
+     * @param  string  $key
+     * @param  mixed  $operator
+     * @param  mixed  $value
+     * @return mixed
+     */
+    public function soleWhere($key, $operator = null, $value = null);
```

Comment:
We can't add methods to an interface mid-cycle.

Remove these from the interface. After it's merged, submit a separate PR to `master` to add them to the interface.

Reply:
That makes sense thinking about it. I'll take this out and make a separate PR if this one gets merged